src/cwv/kpi-metrics.js

```diff
@@ -93,25 +102,29 @@ const calculateKpiDeltasForAuditEntryPerDevice = (entry) => {
  * Calculate aggregated kpiDeltas for all audit entries
  *
  * @param {Object} auditData - Audit data
+ * @param {Object} dataAccess - The data access object for database operations
  * @returns {Object} - Aggregated kpiDeltas for all audit entries
  */
-const calculateKpiDeltasForAudit = (auditData) => {
-  const aggregatedKpiDeltas = {
-    projectedTrafficLost: 0,
-  };
-
-  // Iterate through all entries and aggregate (sum) kpiDeltas
-  auditData.auditResult.cwv.forEach((entry) => {
-    const kpiDeltasForAuditEntryPerDevice = calculateKpiDeltasForAuditEntryPerDevice(
-      entry,
-    );
-
-    Object.values(kpiDeltasForAuditEntryPerDevice).forEach(
-      ({ projectedTrafficLost }) => {
-        aggregatedKpiDeltas.projectedTrafficLost += projectedTrafficLost;
-      },
-    );
-  });
+const calculateKpiDeltasForAudit = (auditData, dataAccess) => {
+  const cpcValue = resolveCpcValue(auditData, dataAccess);
+
+  const aggregatedKpiDeltas = auditData.auditResult.cwv.reduce(
+    (aggregated, entry) => {
+      const kpiDeltasForAuditEntryPerDevice = calculateKpiDeltasForAuditEntryPerDevice(
+        entry,
+        cpcValue,
+      );
+
+      return Object.values(kpiDeltasForAuditEntryPerDevice).reduce(
+        (acc, { projectedTrafficLost, projectedTrafficValue }) => ({
+          projectedTrafficLost: acc.projectedTrafficLost + projectedTrafficLost,
+          projectedTrafficValue: acc.projectedTrafficValue + projectedTrafficValue,
+        }),
+        aggregated,
+      );
+    },
+    { projectedTrafficLost: 0, projectedTrafficValue: 0 },
+  );
```

Comment:
Instead of having nested `reduce` we could add some method chaining here using the map/flatMap/reduce magic:

```js
const calculateKpiDeltasForAudit = (auditData, dataAccess) => {
  const cpcValue = resolveCpcValue(auditData, dataAccess);

  return auditData.auditResult.cwv
    .map(entry => calculateKpiDeltasForAuditEntryPerDevice(entry, cpcValue))
    .flatMap(Object.values)
    .reduce(
      (totals, { projectedTrafficLost, projectedTrafficValue }) => ({
        projectedTrafficLost: totals.projectedTrafficLost + projectedTrafficLost,
        projectedTrafficValue: totals.projectedTrafficValue + projectedTrafficValue,
      }),
      { projectedTrafficLost: 0, projectedTrafficValue: 0 }
    );
};
```